lib/config/configuration.go

```diff
@@ -608,6 +613,12 @@ func Configure(clf *CommandLineFlags, cfg *service.Config) error {
 		cfg.AuthServers = append(cfg.AuthServers, cfg.Auth.SSHAddr)
 	}
 
+	// add data_dir to the backend config:
+	if cfg.Auth.StorageConfig.Params == nil {
+		cfg.Auth.StorageConfig.Params = backend.Params{}
+	}
+	cfg.Auth.StorageConfig.Params["data_dir"] = cfg.DataDir
```

Comment:
@russjones this overwrites the user-supplied value at all times, what should not be the case. We should only overwrite the value if user haven't supplied the value. You can fix that yourself as @kontsevoy is OOO at the moment.